File: ES01-jakarta-startkit/src/main/java/it/distributedsystems/web/HibernateServlet.java
package it.distributedsystems.web;

import it.distributedsystems.util.HibernateUtil;
import jakarta.servlet.ServletException;
import jakarta.servlet.annotation.WebServlet;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import it.distributedsystems.model.ejb.*;
import org.hibernate.Session;
import org.hibernate.query.Query;

import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;

@WebServlet(name = "HibernateServlet", urlPatterns = {"hibernate"}, loadOnStartup = 1) // TODO: demo/hibernate???
public class HibernateServlet extends HttpServlet {

    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {

        initSessionFactory(response);
    }

    private void initSessionFactory(HttpServletResponse response) throws IOException {

        Session session = HibernateUtil.getSessionFactory().openSession();

        session.beginTransaction();

        Department department = new Department("java");
        session.persist(department);

        session.persist(new Employee("Jakab Gipsz",department));
        session.persist(new Employee("Captain Nemo",department));

        session.getTransaction().commit();

        Query q = session.createQuery("From Employee ");

        PrintWriter out = response.getWriter();
        List<Employee> resultList = q.list();
        System.out.println("num of employess:" + resultList.size());
        out.println("num of employees:" + resultList.size());
        for (Employee next : resultList) {
            System.out.println("next employee: " + next);
            out.println(next.toString());
        }

        /*
        session.disconnect();
        session.close();
        HibernateUtil.shutdown();
        */
    }

    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {


        initSessionFactory(response);
    }






}

File: Ex1/src/main/kotlin/it/distributedsystems/web/HibernateServlet.kt
package it.distributedsystems.web

import it.distributedsystems.util.HibernateSessionFactory.sessionFactory
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServlet
import kotlin.Throws
import javax.servlet.ServletException
import java.io.IOException
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import it.distributedsystems.model.ejb.Department
import it.distributedsystems.model.ejb.Employee

@WebServlet(name = "HibernateServlet", urlPatterns = ["hibernate"], loadOnStartup = 1) // TODO: demo/hibernate???
class HibernateServlet : HttpServlet() {
    @Throws(ServletException::class, IOException::class)
    override fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
        initSessionFactory(response)
    }

    @Throws(IOException::class)
    private fun initSessionFactory(response: HttpServletResponse) {
        val session = sessionFactory.openSession()
        session.beginTransaction()
        val department = Department(name = "java")
        session.save(department)
        session.save(Employee(name = "Jakab Gipsz", department = department))
        session.save(Employee(name = "Captain Nemo", department = department))
        session.transaction.commit()
        val q = session.createQuery("From Employee ")
        val out = response.writer
        val resultList = q.list() as List<Employee>
        println("num of employess:" + resultList.size)
        out.println("num of employees:" + resultList.size)
        for (next in resultList) {
            println("next employee: $next")
            out.println(next.toString())
        }

        /*
        session.disconnect();
        session.close();
        HibernateUtil.shutdown();
        */
    }

    @Throws(ServletException::class, IOException::class)
    override fun doPost(request: HttpServletRequest, response: HttpServletResponse) {
        initSessionFactory(response)
    }
}